bids/modeling/statsmodels.py

```diff
@@ -592,25 +587,18 @@ def __init__(self, node, entities={}, collections=None, inputs=None,
 
         var_names = list(self.node.model['x'])
 
-        # Handle the special 1 construct. If it's present, we add a
-        # column of 1's to the design matrix. But behavior varies:
-        # * If there's only a single contrast across all of the inputs,
-        #   the intercept column is given the same name as the input contrast.
-        #   It may already exist, in which case we do nothing.
-        # * Otherwise, we name the column 'intercept'.
-        int_name = None
+        # Handle the special 1 construct.
+        # Add column of 1's to the design matrix called "intercept" 
         if 1 in var_names:
-            if ('contrast' not in df.columns or df['contrast'].nunique() > 1):
-                int_name = 'intercept'
-            else:
-                int_name = df['contrast'].unique()[0]
-
-            var_names.remove(1)
+            var_names = ['intercept' if i == 1 else i for i in var_names]
+            if 'intercept' not in df.columns:
+                df.insert(0, 'intercept', 1)
 
-            if int_name not in df.columns:
-                df.insert(0, int_name, 1)
-            else:
-                var_names.append(int_name)
+        # If a single incoming contrast
+        if ('contrast' in df.columns and df['contrast'].nunique() == 1):
+            unique_in_contrast = df['contrast'].unique()[0]
```

Comment:
this variable is only set if a Node iteration is grouped by `contrast`. 

we later use this to rename contrasts with concatenation